shared/js/background/classes/tracker.es6.js

```diff
@@ -16,9 +17,9 @@ class Tracker {
      * to track you through many different entities.
      * We store a list of all unique urls here.
      */
-    addURL (url) {
-        if (this.urls.indexOf(url) === -1) {
-            this.urls.push(url)
+    update (t) {
+        if (!this.urls[t.url]) {
+            this.urls[t.url] = {isBlocked: t.block, reason: t.reason}
```

Comment:
@MariagraziaAlastra  @jdorweiler There's multple places where we have to call `Object.keys` on this object. Why not store these both as an array and an object?

Reply:
This update method runs in request blocking code so pushing that off to the UX is probably best for performance.

Reply:
Hmm that's fair. Any chance we could just do it once on the UI code, then?

Reply:
@andrey-p how about I add the urls array to the company object in companyListMap

Reply:
@MariagraziaAlastra Yeah that sounds good. Do it!

Reply:
@andrey-p Ok so just writing here what we talked about via dm: I reduced the calls from companyListMap (which also allows us to not call Object.keys() in the tracker networks template) but unfortunately the site model still needs to do that, because we use trackersBlocked and trackers there, not the companyListMap.

I think we'll need to figure out an adequate structure with a followup tech design.

Reply:
@andrey-p - a better way to frame your question is: it is not convenient to iterate over a javascript plain object. I don't think the answer is to store the same information in two ways without a shared interface to it. You either a) punt, b) hide the Object.keys stuff behind another interface, or c) replace it with a better structure, like a Map.

I don't think right now is the time to change this. Object.keys might be slower or inconvenient, but this discussion sounds like it's digging the hole deeper.